src/Microsoft.Health.Dicom.Core/Features/Routing/IUrlResolver.cs

```diff
@@ -45,8 +45,9 @@ public interface IUrlResolver
     /// Resovles the URI to retrieve an instance.
     /// </summary>
     /// <param name="instanceIdentifier">The identifier to the instance.</param>
+    /// <param name="isPartitionEnabled">Whether or not partition feature is enabled</param>
     /// <returns>An instance of <see cref="Uri"/> pointing to where the instance can be retrieved.</returns>
-    Uri ResolveRetrieveInstanceUri(InstanceIdentifier instanceIdentifier);
+    Uri ResolveRetrieveInstanceUri(InstanceIdentifier instanceIdentifier, bool isPartitionEnabled);
```

Comment:
Unsure how I feel about this. I like bubbling up the partition concept to the signature as opposed to the hidden HttpContext resolution. There are concepts like "studyInstanceUid` above that are UIDs without the context of a partition. Should there be an "identifier" class that is agnostic of partition? Or perhaps the better question is: how should the URI resolver consistently across its methods know whether to use a partition? Perhaps that is something determined by DI (e.g. a partitioned "UrlResolver" versus a non-partitioned?)

Reply:
Also would the method above `ResolveRetrieveStudyUri` need partition information?

Reply:
Why do we need this? Already the existing code populates the partition based on route values